mp2-v1/src/values_extraction/mod.rs

```diff
@@ -17,8 +17,10 @@ use std::iter;
 pub mod api;
 mod branch;
 mod extension;
+mod gadgets;
 mod leaf_mapping;
 mod leaf_single;
+mod leaves;
```

Comment:
Why don't replace the `leaf_mapping` and `leaf_single` circuits instead of adding a new module?